core/src/main/java/org/apache/calcite/rel/metadata/RelMdColumnOrigins.java

```diff
@@ -65,8 +65,8 @@ private RelMdColumnOrigins() {}
   public Set<RelColumnOrigin> getColumnOrigins(Aggregate rel,
       RelMetadataQuery mq, int iOutputColumn) {
     if (iOutputColumn < rel.getGroupCount()) {
-      // Group columns pass through directly.
-      return mq.getColumnOrigins(rel.getInput(), iOutputColumn);
+      // get actual index of Group columns.
+      return mq.getColumnOrigins(rel.getInput(), rel.getGroupSet().asList().get(iOutputColumn));
```

Comment:
The title does not describe the issue clearly. Could you improve it?

Reply:
thank you, I will try it.